### PR TITLE
Functioning Station module

### PR DESCRIPTION
**Pull request by SuhasHebbar**
This PR does not incorporate the modified itinerary structure as incorporating the changes to a non functioning Station module would make any source of error ambiguous.

The Station Module is still in need of refactoring to improve code quality and semantics.

In the test for rapid processing of queries the Station module is unable to consume more than 10,000 queries in a second which implies that there is a significant bottleneck to be removed.

*** In the travis build rapid processing of queries fails at 1000 queries but on my system 10,000 queries are processed comfortably hence passing the test.

### CODE REVIEW

**Comment by SuhasHebbar** [comment]
Best case scenario the bottleneck is the mocks used, which would imply the module itself functions fine. I presently plan to quickly make a non FSM version of the present Station module to be used as a baseline for the profiling

This should not take long as event though the structure of the Station was supposed to be a FSM the Events have to triggered manually, hence giving an overall structure similiar to function calls.
```
		station_fsm = StationFsm.query_input(station_fsm, itinerary) |>
		StationFsm.check_query_status

		if(StationFsm.state(station_fsm) != :ready) do
			station_fsm = StationFsm.initialize(station_fsm)
		end

		if(StationFsm.state(station_fsm) != :ready) do
			station_fsm = StationFsm.process_station_schedule(station_fsm)
		end
```

As one can see above there will be a one to one correspondence from the FSM events to functions.

**Comment by prasadtalasila** [comment]
@SuhasHebbar The build fails because of code quality problems raised by credo. Please fix them. Once that is done, work towards pushing the code quality to 100%.

**Comment by SuhasHebbar** [comment]
The code quality issues are mostly trivial dealing with inconsistencies with the line length, brackets and commas. They will be fixed by today itself.

What I was pointing to was that if one runs the tests they will find that the number of queries processed in 1 second caps off at 10,000. Our benchmarks if run for 1 second for a process predict that a station should process around 40,000 messages.

Is this performance within the acceptable limits?

**Comment by prasadtalasila** [comment]
@SuhasHebbar The so called trivial code quality issues are the ones that bring down projects.

Your observation of reduction in performance is to be expected. The station process has functions that take some time where as the benchmarks simply return without doing any work. Thus the station process would give reduced performance.

Obviously 10,000 messages per second is not enough, but let's take up that problem in another PR. All the tests that check for one lakh messages can be set to failure expectation so that they don't fail the travis build.

In summary, solve the code quality issues, increase the test coverage to 100%. Achieving both the tasks would allow us to successfully close this PR.

**Comment by SuhasHebbar** [comment]
The code coverage is presently 99% not 100% as the dummy function ```run``` in profile.ex. But one can see that all other modules have complete test coverage.

**Comment by prasadtalasila** [comment]
@SuhasHebbar The code coverage looks good. There are still a few code quality issues pending. Also the station process is [crashing](https://travis-ci.org/prasadtalasila/TransportScheduler/builds/333208271#L557) during the test. You need to check the reason for this problem.

**Comment by prasadtalasila** [comment]
@SuhasHebbar I checked existing committed code of **ts-v1.0-alpha** branch and codeclimate shows many, many [issues](https://codeclimate.com/github/prasadtalasila/TransportScheduler/issues).  Of course, you are modifying and improving certain files with this PR. Let's make sure that the incoming PR comes clean without any codeclimate issues. In order to facilitate that check, I am going to close this PR and open a new PR with the same source and destination branches. Hopefully the codeclimate check kicks in from now on.

You can setup the codeclimate code quality monitoring for **SuhasHebbar:ts-v1.0-alpha** branch. That would give you an insight into code quality problems even before you open a pull request.
